### ISSUE TITLE
Crash since macOS BigSur update

### ISSUE THREAD

**Issue by choco** [AdguardTeam/AdGuardForSafari#506]
### Issue Details

* Environment name and version: Safari Technology Preview Release 117 (Safari 14.1, WebKit 16611.1.7.2)
* Operating System: macOS BigSur 11.1
* Filters you use in AdGuard: Ad Blocking (base and mobile), Privacy (adguard tracking protection), Social Widgets (adguard social media filter), Annoyances (adguard filter), security (malicious urls, spam404 and nocoin), Language specific (Adguard French and German, Easylist Italy)
* Any other browser extensions you have installed: 1Password, Cascadea, DuckGo, Popcorn, Refined Github, Sessions, Tampermonkey, Turn off the lights, Piper)

* **Version of AdGuard for Safari:**
  * Version 1.8.10
* **Environment name and version:**
  * Safari Technology Preview Release 117 (Safari 14.1, WebKit 16611.1.7.2)
* **Operating system and version:**
  *  macOS BigSur 11.1
* **Filters you use in AdGuard:**
  * Ad Blocking (base and mobile)
  * Privacy (adguard tracking protection)
  * Social Widgets (adguard social media filter)
  * Annoyances (adguard filter)
  * security (malicious urls, spam404 and nocoin)
  * Language specific (Adguard French and German, Easylist Italy)
* **Any other browser extensions you have installed:**
  * 1Password
  * Cascadea
  * DuckDuckGo
  * Popcorn
  * Refined Github
  * Sessions
  * Tampermonkey
  * Turn off the lights, Piper

### Expected Behavior
Safri doesn't crash

### Actual Behavior
Safari crashes

### Additional Information
The crash is not reproducible, it happens randomly after some time (could be 5 minutes, could be 1 hour, etc..)
The culprit seems to be the Advanced Blocking extension. If I disable that one Safari doesn't crash.
The crash report doesn't actually give useful information, since the crashing thread is the main one, every time with a different stack trace. 

Here's also a thread in the apple stack exchange with people reporting the same issue. https://apple.stackexchange.com/questions/410121/safari-keeps-crashing-after-updating-to-big-sur-termination-reason-namespace-a and one from the apple forums https://discussions.apple.com/thread/252054569

**Comment by Javert24601** [comment]
Yes, yes, yes! I second everything Choco describes with a long list of these random crashes while it seems AdGuard may be the reason.

**Comment by ameshkov** [comment]
Tbh, I am not sure what we can do with this. In the linked threads I see different extensions said to be causing it, not just content blockers.

**Comment by windwak3r** [comment]
Please leave a comment if the issue is still relevant.
